packages/core/data-transfer/src/strapi/providers/remote-source/index.ts

```diff
@@ -151,10 +157,26 @@ class RemoteStrapiSourceProvider implements ISourceProvider {
             const chunk = Buffer.from(rawBuffer.data);
 
             await this.writeAsync(assets[item.assetID].stream, chunk);
+            assetStatus[item.assetID] -= 1;
           }
 
           // The asset has been transferred
           else if (action === 'end') {
+            /**
+             * NOTE: This suggests we're processing streaming events wrong.
+             * We need to investigate and delay 'end' from being called
+             * until streaming has finished, rather than waiting here
+             * */
+            // Wait for all chunks to finish streaming before continuing to the end
+            await new Promise<void>((resolve) => {
+              const interval = setInterval(() => {
+                if (assetStatus[item.assetID] === 0) {
```

Comment:
What if `.end` has been called, but one asset gets stuck while transferring? Won't that block the transfer completely?

Reply:
Should I add a check to throw an error if not resolved in x seconds?

Reply:
We should potentially do that, yeah. The only thing is: how does the transfer react in this case?
- Should we delete the asset?
- Or save it anyway?
- How should we report it to the users?

A lot of questions with no easy answers :sweat_smile: